openapi_core/schema/schemas/models.py

```diff
@@ -155,24 +155,42 @@ def get_all_required_properties_names(self):
 
         return set(required)
 
-    def get_cast_mapping(self, custom_formatters=None, strict=True):
+    def get_cast_mapping(self, custom_formatters=None, strict=True,
+                         require_all_props=False):
         primitive_unmarshallers = self.get_primitive_unmarshallers(
-            custom_formatters=custom_formatters)
+            custom_formatters=custom_formatters,
+            require_all_props=require_all_props
+        )
 
         primitive_unmarshallers_partial = dict(
-            (t, functools.partial(u, type_format=self.format, strict=strict))
+            (t, functools.partial(
+                u,
+                type_format=self.format,
+                strict=strict,
+                require_all_props=require_all_props)
+             )
             for t, u in primitive_unmarshallers.items()
         )
 
-        pass_defaults = lambda f: functools.partial(
```

Comment:
I needed to refactor that part since the parameters were not passed in right with that lambda function here.